sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2260,6 +2260,13 @@ object SQLConf {
       .timeConf(TimeUnit.SECONDS)
       .createWithDefault(300L)
 
+  val STATE_STORE_MAINTENANCE_PROCESSING_TIMEOUT =
+    buildConf("spark.sql.streaming.stateStore.maintenanceProcessingTimeout")
+      .internal()
+      .doc("Timeout in seconds to wait for maintenance to process this partition.")
+      .timeConf(TimeUnit.SECONDS)
+      .createWithDefault(300L)
```

Comment:
i think 5 mins is too large. Because this means a maintenance thread would be waiting for 5 mins for a partition. Instead of doing some other work. We can end up having many threads waiting, doing nothing for 5 mins. Lets set this lower cc @anishshri-db